app/src/sharedTest/java/org/oppia/app/activity/InputInteractionViewTestActivityTest.kt

```diff
@@ -0,0 +1,83 @@
+@RunWith(AndroidJUnit4::class)
+class InputInteractionViewTestActivityTest {
+  private lateinit var launchedActivity: Activity
+  @get:Rule
+  var activityTestRule: ActivityTestRule<InputInteractionViewTestActivity> = ActivityTestRule(
+    InputInteractionViewTestActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Test
+  fun testNumericInputInteractionView_withNoInputText_hasCorrectPendingAnswerType() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_input_interaction_view) as NumericInputInteractionView
+      assertThat(textAnswerRetriever.getPendingAnswer(), instanceOf(InteractionObject::class.java))
+    }
+  }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedText_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_input_interaction_view)).perform(ViewActions.clearText(), ViewActions.typeText("9"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_input_interaction_view) as NumericInputInteractionView
+      assertEquals(textAnswerRetriever.getPendingAnswer().objectTypeCase, InteractionObject.ObjectTypeCase.REAL)
+      assertThat(textAnswerRetriever.getPendingAnswer().real, `is`("9".toDouble()))
+    }
+  }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedText_hasCorrectPendingAnswerWithDecimalValues() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_input_interaction_view)).perform(
+      ViewActions.clearText(),
+      ViewActions.typeText("9.5")
+    )
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_input_interaction_view) as NumericInputInteractionView
+      assertEquals(textAnswerRetriever.getPendingAnswer().objectTypeCase, InteractionObject.ObjectTypeCase.REAL)
+      assertThat(textAnswerRetriever.getPendingAnswer().real, `is`(9.5))
+    }
+  }
+
+  @Test
+  fun testNumericInputInteractionView_withInputtedText_onConfigurationChange_hasCorrectPendingAnswer() {
+    Intents.init()
+    val intent = Intent(Intent.ACTION_PICK)
+    launchedActivity = activityTestRule.launchActivity(intent)
+    onView(withId(R.id.test_number_input_interaction_view)).perform(ViewActions.clearText(), ViewActions.typeText("9"))
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+    onView(withId(R.id.test_number_input_interaction_view)).check(matches(isDisplayed())).check(matches(withText("9")))
+    Intents.release()
```

Comment:
this should be in `tearDown` function. Check `SplashActivityTest`

Reply:
resolved